Pegasus/sources/Primitives.cpp

```diff
@@ -0,0 +1,184 @@
+{
+    m_pScene->RemoveShape<arion::Plane>(m_shapeHandle);
+    RemoveObject<arion::Plane>();
+}
+
+arion::Plane& Plane::GetShape() const
+{
+    return m_pScene->GetShape<arion::Plane>(m_shapeHandle);
+}
+
+Sphere::Sphere(Scene& scene, Type type, mechanics::Body body, arion::Sphere sphere)
+    : Primitive(scene, type, body)
+{
+    InitializeShape(sphere, body);
+    m_shapeHandle = m_pScene->MakeShape<arion::Sphere>();
+    m_pScene->GetShape<arion::Sphere>(m_shapeHandle) = sphere;
+    MakeObject<arion::Sphere>();
+}
+
+Sphere::~Sphere()
+{
+    m_pScene->RemoveShape<arion::Sphere>(m_shapeHandle);
+    RemoveObject<arion::Sphere>();
+}
+
+arion::Sphere& Sphere::GetShape() const
+{
+    return m_pScene->GetShape<arion::Sphere>(m_shapeHandle);
+}
+
+Box::Box(Scene& scene, Type type, mechanics::Body body, arion::Box box)
+    : Primitive(scene, type, body)
+{
+    InitializeShape(box, body);
+    m_shapeHandle = m_pScene->MakeShape<arion::Box>();
+    m_pScene->GetShape<arion::Box>(m_shapeHandle) = box;
+    MakeObject<arion::Box>();
+}
+
+Box::~Box()
+{
+    m_pScene->RemoveShape<arion::Box>(m_shapeHandle);
+    RemoveObject<arion::Box>();
+}
+
+arion::Box& Box::GetShape() const
+{
+    return m_pScene->GetShape<arion::Box>(m_shapeHandle);
+}
+
+void PrimitiveGroup::BindForce(ForceBase& force)
+{
+    if (std::find_if(m_forces.begin(), m_forces.end(),
+                        [&force](ForceBase* f) { return &force == f; }) != m_forces.end())
+    {
+        return;
+    }
+
+    m_forces.push_back(&force);
+
+    for (Primitive* primitive : m_primitives)
+    {
+        force.Bind(*primitive);
+    }
+}
+
+void PrimitiveGroup::UnbindForce(ForceBase& force)
+{
+    if (std::find_if(m_forces.begin(), m_forces.end(),
+                        [&force](ForceBase* f) { return &force == f; }) == m_forces.end())
+    {
+        return;
+    }
+
+    for (Primitive* primitive : m_primitives)
+    {
+        force.Unbind(*primitive);
+    }
+
+    m_forces.erase(
+        std::remove_if(m_forces.begin(), m_forces.end(),
+                        [&force](ForceBase* f) { return &force == f; }),
+        m_forces.end()
+    );
+}
+
+void PrimitiveGroup::BindPrimitive(Primitive& primitive)
+{
+    if (std::find_if(m_primitives.begin(), m_primitives.end(),
+                        [&primitive](Primitive* p) { return &primitive == p; }) != m_primitives.end())
+    {
+        return;
+    }
+
+    m_primitives.push_back(&primitive);
+
+    for (ForceBase* force : m_forces)
+    {
+        force->Bind(primitive);
+    }
+}
+
+void PrimitiveGroup::UnbindPrimitive(Primitive& primitive)
+{
+    if (std::find_if(m_primitives.begin(), m_primitives.end(),
+                        [&primitive](Primitive* p) { return &primitive == p; }) == m_primitives.end())
+    {
+        return;
+    }
+
+    for (ForceBase* force : m_forces)
+    {
+        force->Unbind(primitive);
+    }
+
+    m_primitives.erase(
+        std::remove_if(m_primitives.begin(), m_primitives.end(),
```

Comment:
Instead of second search, iterator from first `find_if` could be reused. Storing one iterator is better than search.